### PR TITLE
Add sample Retry packet

### LINKED ISSUE #3374: Add sample Retry packet
The Retry integrity check is different than the protection for other packets, so it would make sense to include an example in the document.  We need code for it in addition to the sample so that it can be updated more readily.

### PR DESCRIPTION
**Pull request by martinthomson**
Can someone please verify my work?  I've written two implementations
here and the values match, but I value independent validation.

Closes #3374.

### CODE REVIEW

**Comment by marten-seemann** [comment]
@martinthomson The value matches what quic-go is calculating. Will add this test vector as a unit test to my implementation.

**Comment by martinthomson** [comment]
Woo!  I'm not as bad at this as I thought.  Thanks @marten-seemann.

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -1973,6 +1973,19 @@ cd32f0b5004d9f5754c4f7f2d1f35cf3 f7116351c92b99c8ae5833225cb51855
 ~~~
 
 
+## Retry
+
+This shows a Retry packet that might be sent in response to the Initial packet
+in {{sample-client-initial}}. The integrity check includes the client-chosen
+connection ID value of 0x8394c8f03e515708, but that value is not transmitted
```
Comment (MikeBishop) [inline draft-ietf-quic-tls.md]:
```suggestion
connection ID value of 0x8394c8f03e515708, but that value is not
```

**Comment by Lekensteyn** [comment]
I can also confirm that it verifies in Wireshark. (Likewise for the samples in Appendix A.)

**Pull request merged**